### ISSUE TITLE
upgrade to datatables v2

### ISSUE THREAD

**Issue by pmenzel** [rstudio/DT#1137]
Hi,

are there plans to move DT to [datatables v2](https://datatables.net/blog/2024/datatables-2)?
The new search API of v2 looks very interesting.

best wishes!

---

By filing an issue to this repo, I promise that

- [ ] I have fully read the issue guide at https://yihui.org/issue/.
- [ ] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('DT')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/DT')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [ ] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by stla** [comment]
That would be interesting, but on the other hand I would be embarrassed if the codes creating a **DT** table with custom JavaScript will no longer work. Maybe a package **DT2** would be more appropriate?..

**Comment by philibe** [comment]
If datatables has a V2 version there is a big risk of breaking changes. A DT2 package would be more appropriate. (I have lot of pages using DT datatables).

**Comment by christophscheuch** [comment]
I agree that a DT2 package is a sensible way forward given the widespread use of DT. Is this something RStudio / Posit might want to spearhead or should we try to make it a community effort, @gadenbuie ?

**Comment by gadenbuie** [comment]
Updating to datatables v2 is not a priority at the moment. After reading through the upgrade release post and notes, it seems like it would bring a lot of breaking changes that aren't offset by new benefits to R users. If there are new features that are exciting/interesting to you, it'd be helpful if you list them here.

To answer your question @christophscheuch, if you're interested in creating DT2, I encourage you to go for it!

**Comment by AlbertoFabbri93** [comment]
Hi, I would really like to use the new [ColumnControl](https://datatables.net/extensions/columncontrol/) extension. I often work with tables containing cell types, cell groups, patients, and other variables, and I frequently need to filter the data based on a combination of several of these variables. With the current search box, this process is quite cumbersome and makes it difficult to narrow down the results to exactly what I’m interested in.

P.S.: I typically use DataTables embedded in reports generated with Quarto in RStudio.

**Comment by DanChaltiel** [comment]
To anyone interested, there is a DT2 package on CRAN: https://cran.r-project.org/web/packages/DT2/index.html
